File: Additional Example/Complete Body/Encrypt - Key from JSON Body.js
// String encryption with AES 128 UTF8
//{"data":"{\"userid\":1}","key":"1234","iv":"12345"}
//encrypt json body with key and iv in the json object

var CryptoJS = require("crypto-js");
const program = require("commander");
const { Buffer } = require('buffer');

program
  .option("-d, --data <data>", "Data to process")
  .parse(process.argv);

const options = program.opts();

//Get Command Line Arguments value
var Requestbody = options.data

//Base64 decode and parse it for JSON
const decodedString = Buffer.from(Requestbody, 'base64').toString('utf8');
var parsedata = JSON.parse(decodedString)

//Fetch Key, IV and plain text from json object
var key = parsedata['key']
var iv = parsedata['iv']
var plaintext = parsedata['data']

//encrpyt the cipher text with the key and iv
var bytes  = CryptoJS.AES.encrypt(plaintext, CryptoJS.enc.Utf8.parse(key),
{	
	keySize: 128 / 8,
	iv:  CryptoJS.enc.Utf8.parse(iv),
    mode: CryptoJS.mode.CBC
});

var originalText = bytes.toString();

//create the updated json object to disply in burp
var output = JSON.stringify({"data":originalText,"key":key,"iv":iv})

//print the output 
console.log(output)

File: Additional Example/Complete Body/Python/Encrypt - Key from JSON Body.py
import argparse
from base64 import b64decode,b64encode
from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
from cryptography.hazmat.primitives.padding import PKCS7
from cryptography.hazmat.backends import default_backend
import json

# Create an argument parser
parser = argparse.ArgumentParser(description='Process data argument')
parser.add_argument('-d', '--data', help='File path with decrypted data + base64 in JSON format')

# Parse the arguments
args = parser.parse_args()


def aes_cbc_encrypt(plaintext, key, iv):
    # Convert the key and IV to bytes
    key = key.encode('utf-8')
    iv = iv.encode('utf-8')

    # Create a Cipher object with AES CBC algorithm and PKCS7 padding
    cipher = Cipher(algorithms.AES(key), modes.CBC(iv), backend=default_backend())
    encryptor = cipher.encryptor()

    # Pad the plaintext using PKCS7
    padder = PKCS7(algorithms.AES.block_size).padder()
    padded_plaintext = padder.update(plaintext.encode('utf-8')) + padder.finalize()

    # Encrypt the padded plaintext
    ciphertext = encryptor.update(padded_plaintext) + encryptor.finalize()

    # Base64 encode the ciphertext
    encoded_ciphertext = b64encode(ciphertext).decode('utf-8')

    # Return the encoded ciphertext
    return encoded_ciphertext


with open(args.data, 'r') as file:    
    content = json.load(file).get("data")
        
jsondata = b64decode(content).decode('utf-8')
data = json.loads(jsondata)

ciphertext = data['user_id']
key = data['key']
iv = data['iv']

decrypted_data = aes_cbc_encrypt(ciphertext, key, iv)
data['user_id'] = decrypted_data
print(json.dumps(data))